custom_sticker_layer/src/main/java/com/photoeditorsdk/android/app/custom/CustomGlStickerLayer.kt

```diff
@@ -0,0 +1,95 @@
+package com.photoeditorsdk.android.app.custom
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import androidx.annotation.MainThread
+import ly.img.android.pesdk.backend.layer.StickerGlLayer
+import ly.img.android.pesdk.backend.model.chunk.recycleAfter
+import ly.img.android.pesdk.backend.model.state.layer.ImageStickerLayerSettings
+import ly.img.android.pesdk.backend.model.state.manager.StateHandler
+import ly.img.android.pesdk.kotlin_extension.mapPointsWith
+import ly.img.android.pesdk.utils.*
+import kotlin.math.roundToInt
+
+// Sticker that DON'T respond on alpha areas.
+class CustomGlStickerLayer(stateHandler: StateHandler, settings: ImageStickerLayerSettings) : StickerGlLayer(stateHandler, settings) {
+
+    companion object {
+        // Alpha of min 70% is recognized as click area.
+        const val ALPHA_THRESHOLD = 0.7f
+
+        // Size of the cache.
+        const val CLICK_CACHE_SIZE_IN_PIXEL = 64
+    }
+
+    /**
+     * Low quality cache version for alpha check.
+     * 1. This is faster
+     * 2. It's more memory efficient
+     * 3. It adds a threshold because the pixels are bigger
+     */
+    private var clickAreaCache: Bitmap = BitmapFactoryUtils.NOTHING_BITMAP
+    private var isClickCacheLoading = false
+
+    @MainThread
+    override fun doRespondOnClick(event: TransformedMotionEvent) : Boolean {
+        val isClickInBitmap = isInOpaqueBitmap(event)
+        if (isClickInBitmap && settings.isInEditMode && settings.hasVariants()) {
+            settings.nextVariant()
+            refresh()
+        }
+        return isClickInBitmap
+    }
+
+    // We hijack internalLoadBitmapCache to know when it's time to load the cache
+    override fun internalLoadBitmapCache(pixelSize: Long, sync: Boolean): Boolean {
+        if (!isClickCacheLoading && clickAreaCache == BitmapFactoryUtils.NOTHING_BITMAP) {
```

Comment:
can we add a check for `!sync` here since this is not needed in the case of exporting..